PolyEngine/Core/Src/EnumUtils.hpp

```diff
@@ -99,8 +97,14 @@ namespace Poly {
 		const T* GetData() const { return Data; }
 
 		//------------------------------------------------------------------------------
-		T& operator[](E idx) { HEAVY_ASSERTE(static_cast<IndexType>(idx) < GetSize(), "Index out of bounds!"); return Data[static_cast<IndexType>(idx)]; }
-		const T& operator[](E idx) const { HEAVY_ASSERTE(static_cast<IndexType>(idx) < GetSize(), "Index out of bounds!"); return Data[static_cast<IndexType>(idx)]; }
+		T& operator[](E enumValue) {
+			return const_cast<T&>(const_cast<const EnumArray&>(*this)[enumValue]); //as advised by Effective C++
```

Comment:
From http://en.cppreference.com/w/cpp/language/const_cast : 

> const_cast makes it possible to form a reference or pointer to non-const type that is actually referring to a const object or a reference or pointer to non-volatile type that is actually referring to a volatile object. Modifying a const object through a non-const access path and referring to a volatile object through a non-volatile glvalue results in **undefined behavior**.

Please don't overuse const_cast and rever this change.